In [15]:
import json
import tweepy
import urllib
import time
import pandas as pd

In [16]:
#Autenticación para utilizar la api.

api_key = "XXXXXXXXXXXXXXXX"
api_secret = "XXXXXXXXXXXXXX"

access_token = "XXXXXXXXXXXXXXXXXXXXXXXXX"
access_secret = "XXXXXXXXXXXXXXXXXXXXXXX"

auth = tweepy.OAuthHandler(api_key,api_secret)
auth.set_access_token(access_token,access_secret)
api = tweepy.API(auth, wait_on_rate_limit=True)

Los access_token y secret se dejaron vacíos para ser completados por el profesor.

Por otro lado, se genera la autenticación para poder conectar con la API.

In [17]:
# Generando la variable user ligada al usuario de Rosalia. Con esta variable se podrán realizar las siguientes tareas solicitadas.
user = api.get_user(id="rosalia")

In [18]:
#¿Cuál es el id de @rosalia en twitter?
print(user.id)


1728956976


Por medio del método id, se obtiene el número de ID asociado al usuario. En este caso el mencionado número es 1728956976

In [41]:
#Lista los nombres de los usuarios a los que @rosalia está siguiendo.

ids = []
for page in tweepy.Cursor(api.friends_ids, screen_name="rosalia").pages():
    ids.extend(page)
    if len(page)>=5000:
        time.sleep(60)
    
len(ids)

55

In [20]:
#¿Cuál de ellos tiene más seguidores?
followers = {}

for i in ids:
  user= api.get_user(id=i)
  followers[user.screen_name] = user.followers_count

followers = pd.Series(followers).sort_values(ascending= False)
print("El following que tiene Rosalía con mayor número de followers es {0} con un total de {1} followers.". format(followers.index[0],\
                                                                                                                   followers.iloc[0]))

El following que tiene Rosalía con mayor número de followers es elonmusk con un total de 77795218 followers.


Genero un diccionario con el ID y el recuento de followers para, por último, generar una serie de pandas y ordenar de forma descendente.

Una vez hecho lo anterior, se obtiene el primer valor de la serie con su índice y correspondiente valor asociado.

In [45]:
#Utiliza la get_users_tweets() para obtener el json completo con los cinco últimos tweets de Rosalia. 
#Muestra por pantalla el día, la hora de creación y el texto.

timeline = api.user_timeline("rosalia", count=5,)
timeline

for i in range(5):
  print("Fecha de creación: {}".format(timeline[i].created_at))
  print("\n Texto: {}".format(timeline[i].text))
  print("-"*20, "Siguiente tweet", "-"*20)

Fecha de creación: 2022-03-13 23:13:01

 Texto: RT @JANN_vt: HA COMENZADO LA #MOTOMAMI WEEK @ROSALIA
-------------------- Siguiente tweet --------------------
Fecha de creación: 2022-03-13 21:57:20

 Texto: RT @RosaliaSources: .@Rosalia hace una lectura de la letra de 'HENTAI' vía Tiktok. https://t.co/hahptNuJF7
-------------------- Siguiente tweet --------------------
Fecha de creación: 2022-03-13 21:14:40

 Texto: 🤪 @nbcsnl https://t.co/tOOILn2FlL
-------------------- Siguiente tweet --------------------
Fecha de creación: 2022-03-13 21:12:33

 Texto: ❤️ @nbcsnl https://t.co/bJHk0XzsrX
-------------------- Siguiente tweet --------------------
Fecha de creación: 2022-03-13 00:06:10

 Texto: RT @RosaliaSources: En 4 horas y 30 minutos comenzará #SNL, donde @Rosalia realizará 2 actuaciones.

Link para verlo en directo: 
https://t…
-------------------- Siguiente tweet --------------------


In [42]:
# Recupera los cinco primeros tweets, a partir de la SteamingClient, 
# desde de la entidad del país/región/ciudad de tu elección y que no sean retweets. 
#Muestra por pantalla el id del usuario que realiza el tweet, la fecha, la hora de creación y el texto. 

from tweepy import Stream, StreamListener

class listener(StreamListener):
  def __init__(self):
    super().__init__()
    self.counter= 0
    self.limit = 5
  def on_data(self,status):
    json_data=json.loads(status)
    print("Fecha de creación: "+str(json_data["created_at"]+"\nUsuario: "+json_data["user"]["screen_name"])+ "\nTexto: "+ json_data["text"]
          + "\n"+"-"*60)
    self.counter += 1
    if self.counter < self.limit:
      return True
    else:
      twitterStream.disconnect()
    def on_error(self,status):
      print(status)

Defino en la clase el contenido a mostrar a la hora de generar el stream. Debajo se corre el stream que va imprimiendo la información requerida para las coordenadas especificadas (Montevideo).

En la clase, limito el contenido a 5 tweets. Por otro lado, en el print de on_data declaro que campos quiero imprimir al correr el stream. Cuando el stream llega al limite seteado (5), el mismo se desconecta.

In [44]:
twitterStream = Stream(auth, listener())
lista = []
twitterStream.filter(locations=[-56.172725,-34.936937,-56.010177,-34.877229],)

Fecha de creación: Mon Mar 14 18:29:09 +0000 2022
 Usuario: lumadruga21
 Texto: Ver a mi hermano o mi primo mal me parte el cora
------------------------------------------------------------
Fecha de creación: Mon Mar 14 18:29:45 +0000 2022
 Usuario: lumadruga21
 Texto: Podría haber estado mejor el piedra..
------------------------------------------------------------
Fecha de creación: Mon Mar 14 18:29:49 +0000 2022
 Usuario: Pelado42857038
 Texto: @TelemundoUY abiertos. Y, finalmente, para florón y remate de tanto privatizar, privatícense los Estados, entrégues… https://t.co/mGlyopV90c
------------------------------------------------------------
Fecha de creación: Mon Mar 14 18:30:01 +0000 2022
 Usuario: besenzoni_
 Texto: @CabeTrust @Alepint Pensé lo mismo.
------------------------------------------------------------
Fecha de creación: Mon Mar 14 18:30:02 +0000 2022
 Usuario: LucasM_UY
 Texto: @cabolo4 @MatiasTM71 @Protagonistas71 @GustavoPenades @lista71 @PNACIONAL @lucascabraluy @vi